polaris-react/src/components/ActionList/components/Item/Item.tsx

```diff
@@ -134,3 +142,58 @@ export function Item({
     </>
   );
 }
+
+export const TruncateText = ({
+  children,
+  position = 'end',
+}: {
+  children: string;
+  position?: 'end' | 'middle';
+}) => {
+  const textRef = useRef<HTMLSpanElement>(null);
+  const [isOverflowing, setIsOverflowing] = useState(false);
+
+  useIsomorphicLayoutEffect(() => {
+    if (textRef.current) {
+      setIsOverflowing(
+        textRef.current.scrollWidth > textRef.current.offsetWidth,
+      );
+    }
+  }, [children]);
+
+  const text = (
+    <span ref={textRef} className={styles.Truncate}>
+      {children}
+    </span>
+  );
+
+  const truncatedText = middleTruncate(children, 40);
```

Comment:
Can we do something like:

`const MIDDLE_TRUNCATION_LENGTH = 40;`

At the top of the file?